rpm/files.go

```diff
@@ -0,0 +1,142 @@
+package rpm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"sync"
+
+	"github.com/quay/claircore"
+	"github.com/quay/zlog"
+	"golang.org/x/sync/singleflight"
+)
+
+// FilesCache is used for concurrent access to the map containing
+// [claircore.Layer] -> map[string]struct{}.
+type filesCache struct {
+	c    map[string]map[string]struct{}
+	refs map[string]int
+	mu   sync.Mutex
+	sf   singleflight.Group
+	// This wg is added so tests can wait for goroutines to finish to
+	// be able to confirm functionality.
```

Comment:
It took me a bit to figure out what's going on here, and it was only after I came back to Ross's review comments that I realized the intention. Could we make it clearer what exactly this WaitGroup is used for? Here's kind of what I'm thinking:
```suggestion
	// wg tracks the gc calls made during get. This is particularly useful
	// during testing, so we can wait on all the gc calls before asserting
	// the cache is indeed empty and all the references were accounted for.
```